src/components/chip/chip.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { newSpecPage } from '@stencil/core/testing';
+import { Chip } from './chip';
+
+describe('limel-chip', () => {
+    let page;
+
+    beforeEach(async () => {
+        page = await newSpecPage({
+            components: [Chip],
+        });
+    });
+
+    it('should render with the correct text', async () => {
+        await page.setContent('<limel-chip text="Test chip"></limel-chip>');
+
+        expect(page.root).toEqualHtml(`
+            <limel-chip text="Test chip">
+                <mock:shadow-root>
+                    <button class="chip" role="button">
+                        <span class="text">Test chip</span>
+                    </button>
+                </mock:shadow-root>
+            </limel-chip>
+        `);
+    });
+
+    it('should render as a link when the link prop is provided', async () => {
+        await page.setContent(
+            '<limel-chip text="Test chip" link="{ href: \'https://example.com\' }"></limel-chip>',
+        );
+
+        expect(page.root).toEqualHtml(`
+            <limel-chip text="Test chip" link="{ href: 'https://example.com' }">
+                <mock:shadow-root>
+                    <a class="chip" href="https://example.com">
+                        <span class="text">Test chip</span>
+                    </a>
+                </mock:shadow-root>
+            </limel-chip>
+        `);
+    });
```

Comment:
don't know how to fix this:

```
 FAIL  src/components/chip/chip.e2e.ts
  ● limel-chip › should render as a link when the link prop is provided

    TypeError: Cannot read properties of null (reading 'getAttribute')

      23 |
      24 |         const element = await page.find('limel-chip >>> a.chip');
    > 25 |         expect(element.getAttribute('href')).toEqual('https://example.com');
         |                        ^
      26 |     });
      27 |
      28 |     it('should render with a badge when the badge prop is provided', async () => {

      at Object.<anonymous> (src/components/chip/chip.e2e.ts:25:24)
      ```

Reply:
Is the shadow root interfering maybe? One of the things it does, apart from namespacing CSS, is hide the whole shadow tree from queries to elements above it.

But you could access it like `el = page.root.querySelector('limel-chip-set').shadowRoot` and then try to find the anchor, maybe? I'm just guessing though 😛

Reply:
I don't know and don't want to know! Freaking tests. Just fix it! 🤣